src/public/pages/examples/path-params.js

```diff
@@ -0,0 +1,42 @@
+import K, * as U from "karet.util"
+import * as L    from "partial.lenses"
+import React     from "karet"
+
+import { PathInput }       from "../../components/restricted-input"
+import { PrettyStringify } from "../../components/pretty-stringify"
+
+const getPagePathRoot =
+  U.pipe( x => U.match( /(^\/[^/]+\/?).*/, x )[ 1 ] )
+
+const subPathGetter = pagePathRoot =>
+  U.pipe( U.replace( pagePathRoot, '' )
+                   , decodeURIComponent
+                   )
+
+const subPathSetter = pagePathRoot =>
+  U.pipe( U.map( U.when( x => x !== '/'
+                       , encodeURIComponent
+                       )
+               )
+        , U.join( '' )
+        , U.concat( pagePathRoot )
+        , U.replace( pagePathRoot + '/', pagePathRoot )
+        )
+
+const _subPathL = pagePathRoot =>
+  L.iso( subPathGetter( pagePathRoot )
+       , subPathSetter( pagePathRoot )
+       )
+const subPathL = U.lift( _subPathL )
+
+const decodeProps =
+  U.mapObjIndexed( decodeURIComponent )
+
+export const PathParams = ( { props, path } ) =>
+  <div>
+    <PathInput
+      type="text"
+      label="Path"
+      value={ U.view( subPathL( getPagePathRoot( path ) ), path ) }/>
+    <PrettyStringify space="2" obj={ decodeProps( props ) } />
```

Comment:
Hmm... Usually I try to avoid implicit conversions like that, but this is definitely  case where the implicit conversion is unlikely to cause major problems.  One thing that implicit conversions like that can hide is bugs where you are just passing something completely wrong to the component like accidentally passing the wrong variable from scope. What would happen then is that it is converted to NaN and then treated as default and that can hide the bug (potentially for a long time).